File: backend/src/main/java/com/aiinterview/dashboard/service/DashboardService.java
package com.aiinterview.dashboard.service;

import com.aiinterview.dashboard.dto.DashboardSummaryResponse;
import com.aiinterview.dashboard.dto.DashboardScoreTrendResponse;
import com.aiinterview.dashboard.dto.DashboardAnalyticsPeriod;
import com.aiinterview.dashboard.dto.DashboardAnalyticsResponse;

import java.util.List;

public interface DashboardService {

    DashboardSummaryResponse getSummary(Long userId);

    List<DashboardScoreTrendResponse> getScoreTrend(Long userId, int limit);

    List<DashboardAnalyticsResponse> getAnalytics(Long userId, DashboardAnalyticsPeriod period, int limit);
}

File: frontend/src/features/dashboard/services/dashboard-service.ts
import type {
  DashboardAnalyticsPeriod,
  DashboardAnalyticsResponse,
  DashboardScoreTrendResponse,
  DashboardSummaryResponse,
  DashboardWeaknessResponse,
} from "@/features/dashboard/types/dashboard";
import { apiFetch } from "@/services/api-client";

export function getDashboardSummary(): Promise<DashboardSummaryResponse> {
  return apiFetch<DashboardSummaryResponse>("/api/v1/dashboard/summary");
}

export function getDashboardScoreTrend(
  limit = 10,
): Promise<DashboardScoreTrendResponse[]> {
  return apiFetch<DashboardScoreTrendResponse[]>(
    `/api/v1/dashboard/score-trend?limit=${limit}`,
  );
}

export function getDashboardAnalytics(
  period: DashboardAnalyticsPeriod,
  limit = 6,
): Promise<DashboardAnalyticsResponse[]> {
  const searchParams = new URLSearchParams({
    period,
    limit: String(limit),
  });

  return apiFetch<DashboardAnalyticsResponse[]>(
    `/api/v1/dashboard/analytics?${searchParams.toString()}`,
  );
}

export function getDashboardWeaknesses(): Promise<DashboardWeaknessResponse> {
  return apiFetch<DashboardWeaknessResponse>("/api/v1/dashboard/weaknesses");
}
